File: src/main/java/pl/allegro/experiments/chi/chiserver/infrastructure/experiments/PermissionsAwareExperimentGetterConfig.java
package pl.allegro.experiments.chi.chiserver.infrastructure.experiments;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import pl.allegro.experiments.chi.chiserver.domain.UserProvider;
import pl.allegro.experiments.chi.chiserver.domain.experiments.ExperimentsRepository;
import pl.allegro.experiments.chi.chiserver.domain.experiments.administration.PermissionsAwareExperimentRepository;

@Configuration
public class PermissionsAwareExperimentGetterConfig {

    @Bean
    PermissionsAwareExperimentRepository permissionsAwareExperimentRepository(
            ExperimentsRepository experimentsRepository,
            UserProvider userProvider) {
        return new PermissionsAwareExperimentRepository(experimentsRepository, userProvider);
    }
}

File: src/main/kotlin/pl/allegro/experiments/chi/chiserver/infrastructure/experiments/PermissionsAwareExperimentGetterConfig.kt
package pl.allegro.experiments.chi.chiserver.infrastructure.experiments

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import pl.allegro.experiments.chi.chiserver.domain.UserProvider
import pl.allegro.experiments.chi.chiserver.domain.experiments.ExperimentsRepository
import pl.allegro.experiments.chi.chiserver.domain.experiments.administration.PermissionsAwareExperimentRepository

@Configuration
class PermissionsAwareExperimentGetterConfig {
    @Bean
    fun permissionsAwareExperimentGetter(experimentsRepository: ExperimentsRepository, userProvider: UserProvider):
            PermissionsAwareExperimentRepository {
        return PermissionsAwareExperimentRepository(experimentsRepository, userProvider)
    }
}